pytradfri/api/aiocoap_api.py

```diff
@@ -101,7 +101,7 @@ def psk(self) -> str | None:
     async def _get_protocol(self) -> Context:
         """Get the protocol for the request."""
         if self._protocol is None:
-            self._protocol = asyncio.create_task(Context.create_client_context())
+            self._protocol = asyncio.ensure_future(Context.create_client_context())
```

Comment:
`asyncio.ensure_future` is a low level asyncio API that we want to avoid. I suggest we use an asyncio.Lock for now to guard the creation via `await` of the protocol.

```py
async def _get_protocol(self) -> Context:
    """Get the protocol for the request."""
    async with protocol_lock:
        if self._protocol is None:
            self._protocol = await Context.create_client_context()

    return self._protocol
```

Reply:
Works just as well, applied; I'd squash when approved.